src/Engine/EmitMSIL/Replication.cs

```diff
@@ -133,5 +130,206 @@ private static List<List<ReplicationGuide>> ConstructRepGuides(string[][] replic
             return new List<List<ReplicationGuide>>();
         }
 
+        private static MethodBase SelectFinalFep(IEnumerable<MethodBase> functionEndPoints, List<object> formalParameters)
+        {
+            // TODO: Determine final function endpoint here based on fitting runtime args to function parameters
+            return functionEndPoints.FirstOrDefault();
+        }
+
+        private static object ExecWithZeroRI(MethodBase finalFep, List<object> formalParameters)
+        {
+            // TODO: CoerceParameters
+            List<object> coercedParameters = formalParameters;//finalFep.CoerceParameters(formalParameters);
+
+            // Testing invoking method without replication
+            object result;
+            if (finalFep.IsStatic)
+            {
+                result = finalFep.Invoke(null, coercedParameters.ToArray());
+            }
+            else
+            {
+                result = finalFep.Invoke(coercedParameters[0], coercedParameters.Skip(1).ToArray());
+            }
+
+            // TODO: PerformReturnTypeCoerce
+            // An explicit call requires return coercion at the return instruction
+            //result = PerformReturnTypeCoerce(finalFep, ret);
+
+            return result;
+        }
+
+        // TODO: Look into array conversion performance 
+        private static IEnumerable<object> convertToEnumerable(object o)
+        {
+            IEnumerable<object> oIEnum;
+            if (o is Array oArr)
+            {
+                oIEnum = oArr.Cast<object>();
+            }
+            else if (o is IEnumerable oEnum)
+            {
+                oIEnum = oEnum.Cast<object>();
+            }
+            else
+            {
+                oIEnum = new List<object>() { o };
+            }
+            return oIEnum;
```

Comment:
all these cast<object>() calls will do boxing and they will probably copy the items too...
Not sure if there is a better way to do this...

Reply:
what about trying to use `Memory<T>`? I  think it has a constructor from an array and the you can use:
MemoryMarshal.ToEnumerable...
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.memorymarshal.toenumerable?view=net-6.0

These types are only useful I guess if the `memory` object is really a view onto the array, which I thought I had read it was at some point...